Orange/widgets/data/owselectrows.py

```diff
@@ -628,20 +626,28 @@ def commit(self):
         self.match_desc = report.describe_data_brief(matching_output)
         self.nonmatch_desc = report.describe_data_brief(non_matching_output)
 
-        self.update_info(matching_output, self.data_out_rows, "Out: ")
+        self.update_output_summary(matching_output)
 
-    def update_info(self, data, lab1, label):
+    def update_input_summary(self, data):
+        self.info.set_input_summary(*self._create_summary(data))
+
+    def update_output_summary(self, data):
+        self.info.set_output_summary(*self._create_summary(data))
+
+    def _create_summary(self, data):
         def sp(s, capitalize=True):
             return s and s or ("No" if capitalize else "no"), "s" * (s != 1)
 
-        if data is None:
-            lab1.setText("")
-        else:
-            lab1.setText(label + "~%s row%s, %s variable%s" %
-                         (sp(data.approx_len()) +
-                          sp(len(data.domain.variables) +
-                             len(data.domain.metas)))
-                        )
+        summary, details = self.info.NoInput, ""
```

Comment:
`summay` needs to be passed as a parameter. 
For output summary `self.info.NoOutput` should be used.